langserver/langserver.py

```diff
@@ -75,12 +79,36 @@ def handle(self, request):
                 })
             log.warning("error handling request %s", request, exc_info=True)
         else:
+            write_response_span = Tracer.start_span("send_response", handler_span)
             self.conn.write_response(request["id"], resp)
+            write_response_span.finish()
+
+        handler_span.finish()
 
     def new_script(self, *args, **kwargs):
         return RemoteJedi(self.fs, self.root_path).new_script(*args, **kwargs)
 
-    def serve_initialize(self, request):
+    @staticmethod
+    def goto_definitions(script, request, parent_span):
+        def_span = Tracer.start_span("Script.goto_definitions", parent_span)
+        try:
+            return script.goto_definitions()
+        except Exception as e:
+            # TODO return these errors using JSONRPC properly. Doing it this way
+            # initially for debugging purposes.
+            log.error("Failed goto_definitions for %s", request, exc_info=True)
```

Comment:
this info would be nice to log to the span. You would automatically get that if you use a `with` inside of the try block.